javascript/ql/lib/semmle/javascript/security/dataflow/XssThroughDomCustomizations.qll

```diff
@@ -216,4 +216,30 @@ module XssThroughDom {
       }
     }
   }
+
+
+  /**
+   * Gets a reference to a value obtained by calling `window.getSelection()`. 
+   * https://developer.mozilla.org/en-US/docs/Web/API/Selection
+   */
+  DataFlow::SourceNode getSelectionCall(DataFlow::TypeTracker t) {
+    t.start() and
+    exists(DataFlow::CallNode call |
+      call = DataFlow::globalVarRef("getSelection").getACall()
+    |
```

Comment:
```suggestion
    exists(DataFlow::CallNode call |
      call = DataFlow::globalVarRef("getSelection").getACall()
      or
      call = DOM::documentRef().getAMemberCall("getSelection")
    |
``` 
Two things: 
 1: The autoformatter was complaining about this line. 
 2: The `getSelection()` method also exists on the `Document` object.  
 
For the last point: I usually check [lib.dom.d.ts](https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts) when I need to check which DOM APIs exist.  
(I also checked that file when reviewing your previous PR).  

Could you also add a test for the `document.getSelection()` method?

Reply:
Sure can. I will work on tests later today.
I am sorry about the auto-format error. Usually I am a formatting freak, don't know what happened here =)
Thanks for the [lib.dom.d.ts](https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts) tip, I had no idea the getSelection method was available through the **document** object as well.

Reply:
> Sure can. I will work on tests later today. 

Great. I'll get the CI to run when you've pushed that.